internal/pkg/filesystem/filesystem.go

```diff
@@ -0,0 +1,330 @@
+package filesystem
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/nhatthm/aferocopy"
+	"github.com/spf13/afero"
+	"go.uber.org/zap"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils"
+)
+
+type fsApi interface {
```

Comment:
`fsApi` backend pre filesystem, ... teraz su tu pritomne `localfs` a `memoryfs`, ... ale dalo by sa jednoducho spravit, aby sa pracovalo napr. priamo s S3, ak by to bolo potrebne.